### ISSUE TITLE
JDK19 serviceability_jvmti_j9_0_FAILED check_vthread_consistency_suspended: vthread is expected to be virtual

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#16751]
Failure link
------------

From [an internal build](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk19_j9_extended.openjdk_ppc64_aix_rerun/1/)(`paix817`):
```
openjdk version "19.0.2" 2023-01-17
IBM Semeru Runtime Open Edition 19.0.2+7 (build 19.0.2+7)
Eclipse OpenJ9 VM 19.0.2+7 (build master-042864b42, JRE 19 AIX ppc64-64-Bit Compressed References 20230117_46 (JIT enabled, AOT enabled)
OpenJ9   - 042864b42
OMR      - b86560818
JCL      - edc950934d based on jdk-19.0.2+7)
```
[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?SDK_RESOURCE=customized&TARGET=extended.openjdk&TEST_FLAG=&UPSTREAM_TEST_JOB_NAME=&DOCKER_REQUIRED=false&ACTIVE_NODE_TIMEOUT=0&VENDOR_TEST_DIRS=&EXTRA_DOCKER_ARGS=&TKG_OWNER_BRANCH=adoptium%3Amaster&OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&PLATFORM=ppc64_aix&GENERATE_JOBS=true&KEEP_REPORTDIR=true&PERSONAL_BUILD=false&DOCKER_REGISTRY_DIR=&ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&RERUN_ITERATIONS=3&SETUP_JCK_RUN=false&DOCKER_REGISTRY_URL_CREDENTIAL_ID=&LABEL=&EXTRA_OPTIONS=&CUSTOMIZED_SDK_URL=+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk19u%2Fjdk19u-aix-ppc64-openj9%2F46%2Fibm-semeru-open-debugimage_ppc64_aix_JDK19U_2023-02-18-17-00.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk19u%2Fjdk19u-aix-ppc64-openj9%2F46%2Fibm-semeru-open-jdk_ppc64_aix_JDK19U_2023-02-18-17-00.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk19u%2Fjdk19u-aix-ppc64-openj9%2F46%2Fibm-semeru-open-jre_ppc64_aix_JDK19U_2023-02-18-17-00.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk19u%2Fjdk19u-aix-ppc64-openj9%2F46%2Fibm-semeru-open-testimage_ppc64_aix_JDK19U_2023-02-18-17-00.tar.gz&BUILD_IDENTIFIER=&JENKINS_KEY=&ADOPTOPENJDK_BRANCH=master&LIGHT_WEIGHT_CHECKOUT=false&USE_JRE=false&ARTIFACTORY_SERVER=na.artifactory.swg-devops&KEEP_WORKSPACE=false&USER_CREDENTIALS_ID=83181e25-eea4-4f55-8b3e-e79615733226&JDK_VERSION=19&DOCKER_REGISTRY_URL=&ITERATIONS=1&VENDOR_TEST_REPOS=&JDK_REPO=https%3A%2F%2Fgithub.com%2Fibmruntimes%2Fopenj9-openjdk-jdk19&JCK_GIT_BRANCH=main&RELEASE_TAG=&OPENJ9_BRANCH=master&OPENJ9_SHA=&JCK_GIT_REPO=&VENDOR_TEST_BRANCHES=&OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&UPSTREAM_JOB_NAME=&CLOUD_PROVIDER=&CUSTOM_TARGET=&VENDOR_TEST_SHAS=&JDK_BRANCH=openj9&LABEL_ADDITION=&ARTIFACTORY_REPO=&ARTIFACTORY_ROOT_DIR=&UPSTREAM_TEST_JOB_NUMBER=&DOCKERIMAGE_TAG=&JDK_IMPL=openj9&TEST_TIME=&SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&AUTO_DETECT=true&SLACK_CHANNEL=%23rt-jenkins&DYNAMIC_COMPILE=false&RELATED_NODES=&ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&APPLICATION_OPTIONS=&CUSTOMIZED_SDK_URL_CREDENTIAL_ID=4e18ffe7-b1b1-4272-9979-99769b68bcc2&ARCHIVE_TEST_RESULTS=false&NUM_MACHINES=&OPENJDK_SHA=&TRSS_URL=http%3A%2F%2Ftrss1.fyre.ibm.com&USE_TESTENV_PROPERTIES=false&BUILD_LIST=openjdk&UPSTREAM_JOB_NUMBER=&STF_OWNER_BRANCH=adoptium%3Amaster&TIME_LIMIT=20&JVM_OPTIONS=&PARALLEL=None) - Change TARGET to run only the failed test targets.

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
[2023-02-18T19:05:43.974Z] variation: Mode150
[2023-02-18T19:05:44.314Z] JVM_OPTIONS:  -XX:+UseCompressedOops 

[2023-02-18T19:10:55.905Z] TEST: serviceability/jvmti/stress/StackTrace/Suspended/GetStackTraceSuspendedStressTest.java

[2023-02-18T19:10:55.908Z] STDERR:
[2023-02-18T19:10:55.908Z] 
[2023-02-18T19:10:55.908Z] Fatal error: Agent: check_vthread_consistency_suspended: vthread is expected to be virtual

[2023-02-18T19:14:21.765Z] serviceability_jvmti_j9_0_FAILED
```

This failure occurred across platforms.
[x86-64_linux](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk19_j9_extended.openjdk_x86-64_linux/40/)
[ppc64le_linux](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk19_j9_extended.openjdk_ppc64le_linux/39/)

FYI @babsingh

**Comment by babsingh** [comment]
@JasonFengJ9 This failure should not block JDK19. So, I will target this issue for JDK20. In the meanwhile, we can disable the test.

The failure is intermittent. There is a slow path in the test case, which isn't always taken. In the slow path, the new JVMTI extension function (`GetVirtualThread`) returns a platform thread instead of a virtual thread, which causes the fatal error.

**Comment by thallium** [comment]
The test seems to pass now

**Comment by babsingh** [comment]
Yes, it was fixed with https://github.com/eclipse-openj9/openj9/pull/16786 and https://github.com/eclipse-openj9/openj9/pull/16787. The only remaining step is to re-enable the test. Then, this issue can be closed.

**Comment by pshipton** [comment]
The fix isn't in the 0.37 branch, so we can expect the jdk19 testing on 0.37 to fail.

**Comment by JasonFengJ9** [comment]
> The fix isn't in the 0.37 branch, so we can expect the jdk19 testing on 0.37 to fail.

Do we want this fix in `0.37`? If not, I will just ignore the failure in JDK19 m2 and later builds.
